sdk/servicebus/Azure.Messaging.ServiceBus/tests/Processor/ProcessorLiveTests.cs

```diff
@@ -513,6 +514,45 @@ Task ProcessMessage(ProcessMessageEventArgs args)
             }
         }
 
+        [Test]
+        public async Task StopProcessingAdheresToTokenSLA()
+        {
+            await using (var scope = await ServiceBusScope.CreateWithQueue(
+                enablePartitioning: false,
+                enableSession: false))
+            {
+                // very long timeout
+                await using var client = CreateClient(tryTimeout: 60);
+                var sender = client.CreateSender(scope.QueueName);
+                await sender.SendMessageAsync(GetMessage());
+                await using var processor = client.CreateProcessor(scope.QueueName, new ServiceBusProcessorOptions
+                {
+                    AutoCompleteMessages = true,
+                });
+                var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+                Task ProcessMessage(ProcessMessageEventArgs args)
+                {
+                    tcs.TrySetResult(true);
+                    return Task.CompletedTask;
+                }
+                processor.ProcessMessageAsync += ProcessMessage;
+                processor.ProcessErrorAsync += ExceptionHandler;
+
+                await processor.StartProcessingAsync();
+                await tcs.Task;
+                await Task.Delay(2000); // better way to do this?
```

Comment:
You could  publish a couple of messages to the queue and then have your message handler set a TCS when all published messages were received.   Here, you could await that TCS which would give you a deterministic point to know that a message had been received and, therefore, the processor tasks were active.  

It would also ensure that we know the queue is empty, since you've already seen all of the messages that you had published.  Stopping under those conditions would have previously taken the `TryTimeout` but should behave differently with these changes.